src/config/loadGitHubAppsFromEnvironment.ts

```diff
@@ -1,36 +1,72 @@
-import { AppAuthStrategyOptions } from '@/types';
+import {
+  AppAuthStrategyOptions,
+  GitHubIssuesSomeoneElseCaresAbout,
+} from '@/types';
 
 export class GitHubApp {
+  num: number;
   org: string;
   auth: AppAuthStrategyOptions;
-  project;
+  project: GitHubIssuesSomeoneElseCaresAbout;
 
-  constructor(org: string, opts) {
-    this.org = org;
-    this.auth = {
-      appId: Number(opts.GH_APP_IDENTIFIER),
-      privateKey: opts.GH_APP_SECRET_KEY.replace(/\\n/g, '\n'),
-    };
-    this.project = {
-      node_id: opts.ISSUES_PROJECT_NODE_ID || 'PVT_kwDOABVQ184AOGW8',
-      product_area_field_id:
-        opts.PRODUCT_AREA_FIELD_ID || 'PVTSSF_lADOABVQ184AOGW8zgJEBno',
-      status_field_id: opts.STATUS_FIELD_ID || 'PVTSSF_lADOABVQ184AOGW8zgI_7g0',
-      response_due_date_field_id:
-        opts.RESPONSE_DUE_DATE_FIELD_ID || 'PVTF_lADOABVQ184AOGW8zgLLxGg',
-    };
+  constructor(obj) {
+    this.num = obj.num;
+    this.org = obj.org;
+    this.auth = obj.auth;
+    this.project = obj.project;
   }
 }
 
-export class GitHubAppsRegistry {
-  apps: Map<string, GitHubApp>;
+class GitHubAppsConfigHelper {
+  configs: Map<number, any>; // much looser typing than apps
 
   constructor() {
-    this.apps = new Map<string, GitHubApp>();
+    this.configs = new Map<number, object>();
   }
 
-  register(app) {
-    this.apps.set(app.org, app);
+  forNumber(num: number): object | undefined {
+    if (!this.configs.has(num)) {
+      this.configs.set(num, { num: num });
+    }
+    return this.configs.get(num);
+  }
+
+  validateAll() {
+    const allErrors = new Object();
+    for (const [org, app] of Object.entries(this.configs)) {
+      const errors = new Array();
+      [
+        'auth.appId',
+        'auth.privateKey',
+        'project.node_id',
+        'project.product_area_field_id',
+        'project.status_field_id',
+        'project.response_due_date_field_id',
+      ].forEach((group_key) => {
+        const [group, key] = group_key.split('.');
+        if (!app[group][key]) {
+          errors.push(`${group}.${key}.`);
+        }
+      });
+      if (errors.length) {
+        allErrors[org] = errors;
+      }
+    }
+    if (Object.keys(allErrors).length) {
+      throw new Error(`Config missing: ${JSON.stringify(allErrors)}`);
+    }
```

Comment:
It'd be good to add some tests for validation failures.

Reply:
Added one in #524, could pull up here if worth it.

Reply:
Not worth it. Since we only parse config if the auth envvars are present (`IDENTIFIER`, `SECRET_KEY`), and we have defaults for the project ones, there's nothing for `validateAll` to do. I've removed it in 5bfc74c605dcb957a85e1792e6c7bcbcc6ebaf98, we can bring it back in #524.